Source/TestDafny/README.md

```diff
@@ -0,0 +1,37 @@
+# TestDafny (a.k.a %testdafny)
+
+This internal, testing-only utility is used to execute a single Dafny source file against each compile target option and assert that the output
```

Comment:
I would call it CheckDafnyCompilersOutput, or DafnyCompilers.Test, and potentially special-case it in the lit-runner instead of making it a separate binary (if we decide that we're dropping the usual lit),

Reply:
As discussed offline: I didn't want to make this only work for the xUnit runner since that would definitely break using `lit`, which some developers are (reasonably) still using since I haven't finished on #2033 yet. Making it an internal-only project we can run with `dotnet run` meant I didn't have to pull that into scope yet. 

I'm going to make the lit substitution key `%testDafnyForEachCompiler` instead which seems like it should satisfy your justifiable naming concern. :)